main.cpp

```diff
@@ -213,191 +334,379 @@ struct Computer
+
+    return isBooted;
+}
+
+bool Computer::runProgram(Drive programDrive, std::string path)
+{
+    std::cout << "Running " << path << "on " << programDrive.brand << std::endl;
+    return true;
+}
+
+bool Computer::crash()
+{
+    bool isCrashed = false;
+    
+    if (operatingSystem != "Solaris") 
+    {
+        isCrashed = ((rand() % 1000000) == 0);
+    }
+
+    return isCrashed;
+}
+
 struct Motorcycle
 {
-    // brand
     std::string brand = "Harley Davidson";
-    // model
     std::string model = "2022 Heritage Classic";
-    // colour
     std::string colour = "Pink";
-    // engine power in hp
     int enginePowerHp = 101;
-    // number of cylinders
     int numCylinders = 2;
+    float speedKilometerPerHour = 0.0f; 
 
-    // 3 things it can do:
-    // accelerate
-    float accelerate(float acceleration); // returns new speed
+    void accelerate(float acceleration); 
 
-    // break (decelerate)
-    float decelerate(float deceleration); // returns new speed
+    void decelerate(float deceleration);
 
-    // make a wheelie
-    bool makeWheelie(); // returns true if the front wheel takes off the ground
+    bool makeWheelie(); 
 };
 
+void Motorcycle::accelerate(float acceleration)
+{
+    speedKilometerPerHour += acceleration;
+}
+
+void Motorcycle::decelerate(float deceleration) 
+{
+    speedKilometerPerHour -= deceleration;
+    if (speedKilometerPerHour < 0)
+    {
+        speedKilometerPerHour = 0;
+    }
+}
+
+bool Motorcycle::makeWheelie() 
+{
+    bool wheelie = false;
+    
+    if (speedKilometerPerHour > 20)
+    {
+        wheelie = true;
+    }
+
+    return wheelie;
+}
+
 struct Turntable
 {
-    // platter rotation speed
     float platterRpm = 32.9f;
-    // tonearm counterweight
     float tonearmCounterweightGrams = 160.0f;
-    // motor voltage
     double motorVoltageVolts = 9.2345;
-    // stylus movement amplitude
     float stylusMovementMicrometers = 2.5f;
-    // speed selection
     int speedSelection = 33;
 
-    // 3 things it can do:
-    // change rotating speed
-    float changeRotatingSpeed(bool isIncreased); // returns the new rotating speed
+    void changeSpeedSelection(bool isIncreased); 
 
-    // rotate vinyl disk
-    float rotateDisk(float targetSpeed); // returns the rotating speed
+    void rotatePlatter(); 
 
-    // output groove amplitude
-    float grooveAmplitude(); // returns the current groove amplitude
+    float grooveAmplitude();
 };
 
+void Turntable::changeSpeedSelection(bool isIncreased)
+{
+    if (isIncreased)
+    {
+        speedSelection = 45;
+    }
+    else
+    {
+        speedSelection = 33;
+    }
+}
+
+void Turntable::rotatePlatter()
+{
+    float speedDifference = platterRpm - static_cast< float >(speedSelection);
+    if (std::abs(speedDifference) > 0.5f)
+    {
+        platterRpm -= 0.1f*(speedDifference);
+    }
+}
+
+float grooveAmplitude()
+{
+    return static_cast<float>(rand()) / static_cast<float>(RAND_MAX); // Amplitude between 0.0f and 1.0f
```

Comment:
Casting has not been covered yet.  
Remove this cast.
Change your code so that no warnings are generated **without** using casting.
- [ ] revise